pkg/build/builder/sti.go

```diff
@@ -211,7 +196,7 @@ func (s *STIBuilder) Build() error {
 	// Reset proxies back to their original value.
 	resetHTTPProxy(originalProxies)
 
-	if push {
+	if len(s.build.Status.OutputDockerImageReference) > 0 {
```

Comment:
as @smarterclayton noticed, since you're always setting this now, this will never be true, which is a problem.

in addition he wasn't a fan of having this fake tag logic in the controller since it's a builder-specific detail (just happens to be specific to two of our builders).

So please move this logic back to the way it was.  However instead of using "noOutputTag" as the dummy tag name, use the builder name, as this will avoid some collision issues that @smarterclayton saw when running simultaneous builds.